mlrun/api/crud/marketplace.py

```diff
@@ -169,12 +182,10 @@ def get_source_catalog(
         else:
             catalog = self._catalogs[source_name]
 
-        result_catalog = MarketplaceCatalog(catalog=[])
+        result_catalog = MarketplaceCatalog(catalog=[], channel=source.spec.channel)
         for item in catalog.catalog:
-            if (
-                (channel is None or item.metadata.channel == channel)
-                and (tag is None or item.metadata.tag == tag)
-                and (version is None or item.metadata.version == version)
+            if (tag is None or item.metadata.tag == tag) and (
+                version is None or item.metadata.version == version
```

Comment:
can you comment in what cases this condition is true? or, more accurate, what is that you are doing here?